src/apps/api/views/submissions.py

```diff
@@ -175,6 +176,12 @@ def upload_submission_scores(request, submission_pk):
 @api_view(('GET',))
 def can_make_submission(request, phase_id):
     phase = get_object_or_404(Phase, id=phase_id)
+    user = get_object_or_404(User, username=request.user)
+    user_is_approved = phase.competition.participants.filter(user=request.user, status=CompetitionParticipant.APPROVED).exists()
+
+    if user.is_bot and phase.competition.allow_robot_submissions and not user_is_approved:
+        new_participant = CompetitionParticipant(user=request.user, competition=phase.competition, status=CompetitionParticipant.APPROVED)
+        new_participant.save()
 
     if phase.competition.participants.filter(user=request.user, status=CompetitionParticipant.APPROVED).exists():
```

Comment:
what you're doing above is happening here already, we should have an "else" condition where we do this if we don't already match -- checking for bots is probably much rarer than normal submission. We get ~200-500 submissions per day normally and probably very few of those are bots.